### PR TITLE
Trust builder command

### LINKED ISSUE #566: New command: pack trust-builder
### Description
This is in support of [RFC 25](https://github.com/buildpacks/rfcs/blob/master/text/0025-dont-trust-builders.md) which introduces the concept of trusted builders.

We need a way to tell pack that a builder should be trusted, so that the user does not need to supply `--trust-builder` for every build.

### Proposed solution
**AC 1a**
When I run `pack trust-builder some-build-image`
And I `cat ~/.pack/config.toml`
Then I should see a section in the output corresponding to trusted builders
And I should see that `some-build-image` is there

**AC 1b**
When I run `pack trust-builder some-build-image`
And I run `pack build -B some-build-image --publish`
Then I should see in the logs that the `creator` binary was used

**AC 2** (default builders should be auto-trusted)
When I run `pack build -B cloudfoundry/cnb:bionic --publish`(or any of the suggested builders)
Then I should see in the logs that the `creator` binary was used

### Questions
Do we need to provide a way to un-trust builders?

### PR DESCRIPTION
**Pull request by natalieparellano**
Resolves #566.

### CODE REVIEW

**Comment by natalieparellano** [comment]
I didn't add an acceptance test for this. I am not sure if I should. My thinking:

* An acceptance test that runs `trust-builder` and then prints out the config would be very similar to what is unit tested in trust_builder_test.go
* An acceptance test that runs `trust-builder` and then verifies that the creator is used when `pack build` is run with `--publish` is possible, but then (assuming we trust the builder that is created in the test setup) we lose the benefit of seeing `pack build` run with the 5 phases
  * At one point I considered adding a `TEST_TRUST_BUILDER` environment variable that, if set, would run `trust-builder` on the test builder before running all the other tests. Maybe this would be helpful. But I don't want to explode our test matrix any more than we have to.

**Comment by jromero** [comment]
#### Untrusted builder

##### Output

```
$ ./out/pack build my-app-566 -B sample-builder -p ~/dev/buildpacks/samples/apps/bash-script/ -v
Using project descriptor located at /Users/javier.romero/dev/buildpacks/samples/apps/bash-script/project.toml
Selected run image cnbs/sample-stack-run:alpine
Pulling image cnbs/sample-stack-run:alpine
alpine: Pulling from cnbs/sample-stack-run
Digest: sha256:ff5952abb5476176ddd68790c1ba69ef0114f701c0565acc94ad067100571449
Status: Image is up to date for cnbs/sample-stack-run:alpine
Adding buildpack samples/bash-script version 0.0.1 to builder
Setting custom order
Saving builder with the following buildpacks:
-> samples/java-maven@0.0.1
-> samples/kotlin-gradle@0.0.1
-> samples/hello-world@0.0.1
-> samples/bash-script@0.0.1
Using build cache volume pack-cache-477bee8af0a7.build
===> CREATING
[creator] ---> DETECTING
[creator] ======== Output: samples/bash-script@0.0.1 ========
[creator] ---> Hello Bash Script buildpack
[creator] 
[creator] Here are the contents of the current working directory:
[creator] .:
[creator] total 16
[creator] drwxr-xr-x    2 cnb      cnb           4096 May 20 06:00 .
[creator] drwxr-xr-x    1 root     root          4096 May 20 06:00 ..
[creator] -rwxr-xr-x    1 cnb      cnb            738 Apr  6 23:13 app.sh
[creator] -rw-r--r--    1 cnb      cnb            202 Apr  6 23:13 project.toml
[creator] ======== Results ========
[creator] pass: samples/bash-script@0.0.1
[creator] Resolving plan... (try #1)
[creator] samples/bash-script 0.0.1
[creator] ---> ANALYZING
[creator] Previous image with name "index.docker.io/library/my-app-566:latest" not found
[creator] ---> RESTORING
[creator] ---> BUILDING
[creator] ---> Hello World buildpack
[creator] ---> Hello Bash Script buildpack
[creator] 
[creator] Here are the contents of the current working directory:
[creator] .:
[creator] total 16
[creator] drwxr-xr-x    2 cnb      cnb           4096 May 20 06:00 .
[creator] drwxr-xr-x    1 root     root          4096 May 20 06:00 ..
[creator] -rwxr-xr-x    1 cnb      cnb            738 Apr  6 23:13 app.sh
[creator] -rw-r--r--    1 cnb      cnb            202 Apr  6 23:13 project.toml
[creator] ---> EXPORTING
[creator] no project metadata found at path './project-metadata.toml', project metadata will not be exported
[creator] Writing tarball for layer "launcher"
[creator] Adding layer 'launcher'
[creator] Layer 'launcher' SHA: sha256:32608bc6d97e0193edb0360555b3e08dc6dfe1187833d8548cdd9e662213935b
[creator] Layer 'app' SHA: sha256:4c6ff16e28deaf437a1abe5c926cdfb8afcd2c31ea544f6992ed7b665965b3f4
[creator] Adding 1/1 app layer(s)
[creator] Writing tarball for layer "config"
[creator] Adding layer 'config'
[creator] Layer 'config' SHA: sha256:ed89d623eb7cf47a03403af0b6d3723a967ca4f73985b97056246193d29eb3b0
[creator] *** Images (68706b2783e5):
[creator]       index.docker.io/library/my-app-566:latest
[creator] 
[creator] *** Image ID: 68706b2783e5157738e2a0a009430aabaca7dedb8ec6cc5e824194d7ffe3e99d
Successfully built image my-app-566

$ ./out/pack build my-app-566 -B sample-builder -p ~/dev/buildpacks/samples/apps/bash-script/ -v --publish
Using project descriptor located at /Users/javier.romero/dev/buildpacks/samples/apps/bash-script/project.toml
Selected run image cnbs/sample-stack-run:alpine
Adding buildpack samples/bash-script version 0.0.1 to builder
Setting custom order
Saving builder with the following buildpacks:
-> samples/java-maven@0.0.1
-> samples/kotlin-gradle@0.0.1
-> samples/hello-world@0.0.1
-> samples/bash-script@0.0.1
Pulling image buildpacksio/lifecycle:0.7.5
0.7.5: Pulling from buildpacksio/lifecycle
Digest: sha256:246e6c365f53306583286bcb307b6dd7700e66188d4e9aabf1e4d3139af0f8e1
Status: Image is up to date for buildpacksio/lifecycle:0.7.5
Using build cache volume pack-cache-477bee8af0a7.build
===> DETECTING
[detector] ======== Output: samples/bash-script@0.0.1 ========
[detector] ---> Hello Bash Script buildpack
[detector] 
[detector] Here are the contents of the current working directory:
[detector] .:
[detector] total 16
[detector] drwxr-xr-x    2 cnb      cnb           4096 May 20 06:00 .
[detector] drwxr-xr-x    1 root     root          4096 May 20 06:00 ..
[detector] -rwxr-xr-x    1 cnb      cnb            738 Apr  6 23:13 app.sh
[detector] -rw-r--r--    1 cnb      cnb            202 Apr  6 23:13 project.toml
[detector] ======== Results ========
[detector] pass: samples/bash-script@0.0.1
[detector] Resolving plan... (try #1)
[detector] samples/bash-script 0.0.1
===> ANALYZING
[analyzer] Previous image with name "index.docker.io/library/my-app-566:latest" not found
===> RESTORING
===> BUILDING
[builder] ---> Hello World buildpack
[builder] ---> Hello Bash Script buildpack
[builder] 
[builder] Here are the contents of the current working directory:
[builder] .:
[builder] total 16
[builder] drwxr-xr-x    2 cnb      cnb           4096 May 20 06:00 .
[builder] drwxr-xr-x    1 root     root          4096 May 20 06:00 ..
[builder] -rwxr-xr-x    1 cnb      cnb            738 Apr  6 23:13 app.sh
[builder] -rw-r--r--    1 cnb      cnb            202 Apr  6 23:13 project.toml
===> EXPORTING
[exporter] no project metadata found at path './project-metadata.toml', project metadata will not be exported
[exporter] Writing tarball for layer "launcher"
[exporter] Adding layer 'launcher'
[exporter] Layer 'launcher' SHA: sha256:32608bc6d97e0193edb0360555b3e08dc6dfe1187833d8548cdd9e662213935b
[exporter] Layer 'app' SHA: sha256:4c6ff16e28deaf437a1abe5c926cdfb8afcd2c31ea544f6992ed7b665965b3f4
[exporter] Adding 1/1 app layer(s)
[exporter] Writing tarball for layer "config"
[exporter] Adding layer 'config'
[exporter] Layer 'config' SHA: sha256:9790af0612365f9299ec555ed3b78aa1d71aeca28676b9dc86ba67934f629fbc
[exporter] *** Images (sha256:1b2790f001373b10c08425ba289b0d1bb5674299dbdc27ee81562fb0121447b2):
[exporter]       index.docker.io/library/my-app-566:latest - HEAD https://index.docker.io/v2/library/my-app-566/blobs/sha256:4167d3e149762ea326c26fc2fd4e36fdeb7d4e639408ad30f37b8f25ac285a98: unsupported status code 401
[exporter] 
[exporter] *** Digest: sha256:1b2790f001373b10c08425ba289b0d1bb5674299dbdc27ee81562fb0121447b2
```

#### Untrusted builder with `--trust-builder`

##### Output
```
$ ./out/pack build my-app-566 -B sample-builder -p ~/dev/buildpacks/samples/apps/bash-script/ -v --publish --trust-builder
Using project descriptor located at /Users/javier.romero/dev/buildpacks/samples/apps/bash-script/project.toml
Selected run image cnbs/sample-stack-run:alpine
Adding buildpack samples/bash-script version 0.0.1 to builder
Setting custom order
Saving builder with the following buildpacks:
-> samples/java-maven@0.0.1
-> samples/kotlin-gradle@0.0.1
-> samples/hello-world@0.0.1
-> samples/bash-script@0.0.1
Using build cache volume pack-cache-477bee8af0a7.build
===> CREATING
[creator] ---> DETECTING
[creator] ======== Output: samples/bash-script@0.0.1 ========
[creator] ---> Hello Bash Script buildpack
[creator] 
[creator] Here are the contents of the current working directory:
[creator] .:
[creator] total 16
[creator] drwxr-xr-x    2 cnb      cnb           4096 May 20 06:01 .
[creator] drwxr-xr-x    1 root     root          4096 May 20 06:01 ..
[creator] -rwxr-xr-x    1 cnb      cnb            738 Apr  6 23:13 app.sh
[creator] -rw-r--r--    1 cnb      cnb            202 Apr  6 23:13 project.toml
[creator] ======== Results ========
[creator] pass: samples/bash-script@0.0.1
[creator] Resolving plan... (try #1)
[creator] samples/bash-script 0.0.1
[creator] ---> ANALYZING
[creator] Previous image with name "index.docker.io/library/my-app-566:latest" not found
[creator] ---> RESTORING
[creator] ---> BUILDING
[creator] ---> Hello World buildpack
[creator] ---> Hello Bash Script buildpack
[creator] 
[creator] Here are the contents of the current working directory:
[creator] .:
[creator] total 16
[creator] drwxr-xr-x    2 cnb      cnb           4096 May 20 06:01 .
[creator] drwxr-xr-x    1 root     root          4096 May 20 06:01 ..
[creator] -rwxr-xr-x    1 cnb      cnb            738 Apr  6 23:13 app.sh
[creator] -rw-r--r--    1 cnb      cnb            202 Apr  6 23:13 project.toml
[creator] ---> EXPORTING
[creator] no project metadata found at path './project-metadata.toml', project metadata will not be exported
[creator] Writing tarball for layer "launcher"
[creator] Adding layer 'launcher'
[creator] Layer 'launcher' SHA: sha256:32608bc6d97e0193edb0360555b3e08dc6dfe1187833d8548cdd9e662213935b
[creator] Layer 'app' SHA: sha256:4c6ff16e28deaf437a1abe5c926cdfb8afcd2c31ea544f6992ed7b665965b3f4
[creator] Adding 1/1 app layer(s)
[creator] Writing tarball for layer "config"
[creator] Adding layer 'config'
[creator] Layer 'config' SHA: sha256:ed89d623eb7cf47a03403af0b6d3723a967ca4f73985b97056246193d29eb3b0
[creator] *** Images (sha256:b9279b3ca789074ae90a65492a08d60ef3300f7cb77fc7b41da03b400bcf92a3):
[creator]       index.docker.io/library/my-app-566:latest - HEAD https://index.docker.io/v2/library/my-app-566/blobs/sha256:0defd2e00834fd461fd8652106bd1bd0d4374714318b135de06b1e57d79683d1: unsupported status code 401
[creator] 
[creator] *** Digest: sha256:b9279b3ca789074ae90a65492a08d60ef3300f7cb77fc7b41da03b400bcf92a3
[creator] ERROR: failed to export: failed to write image to the following tags: [index.docker.io/library/my-app-566:latest: HEAD https://index.docker.io/v2/library/my-app-566/blobs/sha256:0defd2e00834fd461fd8652106bd1bd0d4374714318b135de06b1e57d79683d1: unsupported status code 401]
ERROR: failed with status code: 1
┌─[javier.romero][Javiers-MacBook-Pro][±][feature/566-trust-builder → origin {10} ✓][~/dev/buildpacks/pack]
└─▪ ./out/pack build my-app-566 -B sample-builder -p ~/dev/buildpacks/samples/apps/bash-script/ -v --publish
Using project descriptor located at /Users/javier.romero/dev/buildpacks/samples/apps/bash-script/project.toml
Selected run image cnbs/sample-stack-run:alpine
Adding buildpack samples/bash-script version 0.0.1 to builder
Setting custom order
Saving builder with the following buildpacks:
-> samples/java-maven@0.0.1
-> samples/kotlin-gradle@0.0.1
-> samples/hello-world@0.0.1
-> samples/bash-script@0.0.1
Pulling image buildpacksio/lifecycle:0.7.5
0.7.5: Pulling from buildpacksio/lifecycle
Digest: sha256:246e6c365f53306583286bcb307b6dd7700e66188d4e9aabf1e4d3139af0f8e1
Status: Image is up to date for buildpacksio/lifecycle:0.7.5
Using build cache volume pack-cache-477bee8af0a7.build
===> DETECTING
[detector] ======== Output: samples/bash-script@0.0.1 ========
[detector] ---> Hello Bash Script buildpack
[detector] 
[detector] Here are the contents of the current working directory:
[detector] .:
[detector] total 16
[detector] drwxr-xr-x    2 cnb      cnb           4096 May 20 06:01 .
[detector] drwxr-xr-x    1 root     root          4096 May 20 06:01 ..
[detector] -rwxr-xr-x    1 cnb      cnb            738 Apr  6 23:13 app.sh
[detector] -rw-r--r--    1 cnb      cnb            202 Apr  6 23:13 project.toml
[detector] ======== Results ========
[detector] pass: samples/bash-script@0.0.1
[detector] Resolving plan... (try #1)
[detector] samples/bash-script 0.0.1
===> ANALYZING
[analyzer] Previous image with name "index.docker.io/library/my-app-566:latest" not found
===> RESTORING
===> BUILDING
[builder] ---> Hello World buildpack
[builder] ---> Hello Bash Script buildpack
[builder] 
[builder] Here are the contents of the current working directory:
[builder] .:
[builder] total 16
[builder] drwxr-xr-x    2 cnb      cnb           4096 May 20 06:01 .
[builder] drwxr-xr-x    1 root     root          4096 May 20 06:01 ..
[builder] -rwxr-xr-x    1 cnb      cnb            738 Apr  6 23:13 app.sh
[builder] -rw-r--r--    1 cnb      cnb            202 Apr  6 23:13 project.toml
===> EXPORTING
[exporter] no project metadata found at path './project-metadata.toml', project metadata will not be exported
[exporter] Writing tarball for layer "launcher"
[exporter] Adding layer 'launcher'
[exporter] Layer 'launcher' SHA: sha256:32608bc6d97e0193edb0360555b3e08dc6dfe1187833d8548cdd9e662213935b
[exporter] Layer 'app' SHA: sha256:4c6ff16e28deaf437a1abe5c926cdfb8afcd2c31ea544f6992ed7b665965b3f4
[exporter] Adding 1/1 app layer(s)
[exporter] Writing tarball for layer "config"
[exporter] Adding layer 'config'
[exporter] Layer 'config' SHA: sha256:9790af0612365f9299ec555ed3b78aa1d71aeca28676b9dc86ba67934f629fbc
[exporter] *** Images (sha256:1b2790f001373b10c08425ba289b0d1bb5674299dbdc27ee81562fb0121447b2):
[exporter]       index.docker.io/library/my-app-566:latest - HEAD https://index.docker.io/v2/library/my-app-566/blobs/sha256:0defd2e00834fd461fd8652106bd1bd0d4374714318b135de06b1e57d79683d1: unsupported status code 401
[exporter] 
[exporter] *** Digest: sha256:1b2790f001373b10c08425ba289b0d1bb5674299dbdc27ee81562fb0121447b2
[exporter] ERROR: failed to export: failed to write image to the following tags: [index.docker.io/library/my-app-566:latest: HEAD https://index.docker.io/v2/library/my-app-566/blobs/sha256:0defd2e00834fd461fd8652106bd1bd0d4374714318b135de06b1e57d79683d1: unsupported status code 401]
ERROR: failed with status code: 1
```

#### Suggested builder is trusted by default

##### Output
```
./out/pack suggest-builders
Suggested builders:
        Google:                gcr.io/buildpacks/builder                    GCP Builder for all runtimes                                                                   
        Heroku:                heroku/buildpacks:18                         heroku-18 base image with buildpacks for Ruby, Java, Node.js, Python, Golang, & PHP            
        Paketo Buildpacks:     gcr.io/paketo-buildpacks/builder:base        Ubuntu bionic base image with buildpacks for Java, NodeJS and Golang                           
        Paketo Buildpacks:     gcr.io/paketo-buildpacks/builder:full-cf     cflinuxfs3 base image with buildpacks for Java, .NET, NodeJS, Golang, PHP, HTTPD and NGINX     
        Paketo Buildpacks:     gcr.io/paketo-buildpacks/builder:tiny        Tiny base image (bionic build image, distroless run image) with buildpacks for Golang          

Tip: Learn more about a specific builder with:
        pack inspect-builder <builder-image>

$ ./out/pack build my-app-566 -B gcr.io/paketo-buildpacks/builder:base -p ~/dev/buildpacks/samples/apps/bash-script/ -v --publish
Using project descriptor located at /Users/javier.romero/dev/buildpacks/samples/apps/bash-script/project.toml
Pulling image gcr.io/paketo-buildpacks/builder:base
base: Pulling from paketo-buildpacks/builder
Digest: sha256:35566a7b35b1dc5f4efaa7061c6b861b8e0703f4cc8f22d3219d535d19f5acb9
Status: Image is up to date for gcr.io/paketo-buildpacks/builder:base
Selected run image gcr.io/paketo-buildpacks/run:base-cnb
Adding buildpack samples/bash-script version 0.0.1 to builder
Setting custom order
Saving builder with the following buildpacks:
-> paketo-buildpacks/nginx@v0.0.140
-> paketo-buildpacks/dotnet-core@v0.0.1
-> paketo-buildpacks/go@v0.0.1
-> paketo-buildpacks/httpd@v0.0.130
-> paketo-buildpacks/nodejs@v0.0.1
-> paketo-buildpacks/dotnet-core-conf@0.0.122
-> paketo-buildpacks/dotnet-core-runtime@0.0.135
-> paketo-buildpacks/dotnet-core-sdk@0.0.133
-> paketo-buildpacks/icu@0.0.52
-> paketo-buildpacks/node-engine@0.0.178
-> paketo-buildpacks/dotnet-core-aspnet@0.0.128
-> paketo-buildpacks/dotnet-core-build@0.0.70
-> paketo-buildpacks/dep@0.0.109
-> paketo-buildpacks/go-compiler@0.0.112
-> paketo-buildpacks/go-mod@0.0.96
-> paketo-buildpacks/node-engine@0.0.178
-> paketo-buildpacks/npm@0.1.11
-> paketo-buildpacks/yarn-install@0.1.19
-> paketo-buildpacks/java@1.7.1
-> paketo-buildpacks/debug@1.2.4
-> paketo-buildpacks/image-labels@1.0.4
-> paketo-buildpacks/jmx@1.1.4
-> paketo-buildpacks/maven@1.4.1
-> paketo-buildpacks/bellsoft-liberica@2.5.4
-> paketo-buildpacks/google-stackdriver@1.1.4
-> paketo-buildpacks/spring-boot@1.5.4
-> paketo-buildpacks/azure-application-insights@1.1.4
-> paketo-buildpacks/dist-zip@1.3.1
-> paketo-buildpacks/executable-jar@1.2.4
-> paketo-buildpacks/gradle@1.2.1
-> paketo-buildpacks/procfile@1.3.4
-> paketo-buildpacks/sbt@1.2.1
-> paketo-buildpacks/apache-tomcat@1.1.4
-> paketo-buildpacks/encrypt-at-rest@1.2.4
-> samples/bash-script@0.0.1
Using build cache volume pack-cache-477bee8af0a7.build
===> CREATING
[creator] ---> DETECTING
[creator] ======== Output: samples/bash-script@0.0.1 ========
[creator] ---> Hello Bash Script buildpack
[creator] 
[creator] Here are the contents of the current working directory:
[creator] .:
[creator] total 16
[creator] drwxr-xr-x 2 cnb  cnb  4096 May 20 06:06 .
[creator] drwxr-xr-x 1 root root 4096 May 20 06:06 ..
[creator] -rwxr-xr-x 1 cnb  cnb   738 Apr  6 23:13 app.sh
[creator] -rw-r--r-- 1 cnb  cnb   202 Apr  6 23:13 project.toml
[creator] ======== Results ========
[creator] pass: samples/bash-script@0.0.1
[creator] Resolving plan... (try #1)
[creator] samples/bash-script 0.0.1
[creator] ---> ANALYZING
[creator] Previous image with name "index.docker.io/library/my-app-566:latest" not found
[creator] ---> RESTORING
[creator] ---> BUILDING
[creator] ---> Hello World buildpack
[creator] ---> Hello Bash Script buildpack
[creator] 
[creator] Here are the contents of the current working directory:
[creator] .:
[creator] total 16
[creator] drwxr-xr-x 2 cnb  cnb  4096 May 20 06:06 .
[creator] drwxr-xr-x 1 root root 4096 May 20 06:06 ..
[creator] -rwxr-xr-x 1 cnb  cnb   738 Apr  6 23:13 app.sh
[creator] -rw-r--r-- 1 cnb  cnb   202 Apr  6 23:13 project.toml
[creator] ---> EXPORTING
[creator] no project metadata found at path './project-metadata.toml', project metadata will not be exported
[creator] Writing tarball for layer "launcher"
[creator] Adding layer 'launcher'
[creator] Layer 'launcher' SHA: sha256:b7a75fbcf275cde98ff92d00867050e8c56a31e006c234281a3fd36ce4323eb5
[creator] Layer 'app' SHA: sha256:f0d803f25a0f56c10270c762ac7a613252abbbda9f366cbae5958119bebfd87d
[creator] Adding 1/1 app layer(s)
[creator] Writing tarball for layer "config"
[creator] Adding layer 'config'
[creator] Layer 'config' SHA: sha256:112cb236eeb32f71ff12f092cad32ec6f2f72790051b4d055d1f982b19854622
[creator] *** Images (sha256:79991df510dde5777bb966bc615fbbcfc7ba93e0553e9bbbcfbd749d6cccf92b):
[creator]       index.docker.io/library/my-app-566:latest - HEAD https://index.docker.io/v2/library/my-app-566/blobs/sha256:83d7734c59c16c722d3e6f52c52f7672be31f21f153a826edd6d4b892dbddb44: unsupported status code 401
[creator] 
[creator] *** Digest: sha256:79991df510dde5777bb966bc615fbbcfc7ba93e0553e9bbbcfbd749d6cccf92b
```

**Comment by jromero** [comment]
### `pack trust-builder`

##### Output

```
$ cat ~/.pack/config.toml 

$ ./out/pack build my-app-566 -B sample-builder -p ~/dev/buildpacks/samples/apps/bash-script/ -v --publish
Using project descriptor located at /Users/javier.romero/dev/buildpacks/samples/apps/bash-script/project.toml
Selected run image cnbs/sample-stack-run:alpine
Adding buildpack samples/bash-script version 0.0.1 to builder
Setting custom order
Saving builder with the following buildpacks:
-> samples/java-maven@0.0.1
-> samples/kotlin-gradle@0.0.1
-> samples/hello-world@0.0.1
-> samples/bash-script@0.0.1
Pulling image buildpacksio/lifecycle:0.7.5
0.7.5: Pulling from buildpacksio/lifecycle
Digest: sha256:246e6c365f53306583286bcb307b6dd7700e66188d4e9aabf1e4d3139af0f8e1
Status: Image is up to date for buildpacksio/lifecycle:0.7.5
Using build cache volume pack-cache-477bee8af0a7.build
===> DETECTING
[detector] ======== Output: samples/bash-script@0.0.1 ========
[detector] ---> Hello Bash Script buildpack
[detector] 
[detector] Here are the contents of the current working directory:
[detector] .:
[detector] total 16
[detector] drwxr-xr-x    2 cnb      cnb           4096 May 20 06:23 .
[detector] drwxr-xr-x    1 root     root          4096 May 20 06:23 ..
[detector] -rwxr-xr-x    1 cnb      cnb            738 Apr  6 23:13 app.sh
[detector] -rw-r--r--    1 cnb      cnb            202 Apr  6 23:13 project.toml
[detector] ======== Results ========
[detector] pass: samples/bash-script@0.0.1
[detector] Resolving plan... (try #1)
[detector] samples/bash-script 0.0.1
===> ANALYZING
[analyzer] Previous image with name "index.docker.io/library/my-app-566:latest" not found
===> RESTORING
===> BUILDING
[builder] ---> Hello World buildpack
[builder] ---> Hello Bash Script buildpack
[builder] 
[builder] Here are the contents of the current working directory:
[builder] .:
[builder] total 16
[builder] drwxr-xr-x    2 cnb      cnb           4096 May 20 06:23 .
[builder] drwxr-xr-x    1 root     root          4096 May 20 06:24 ..
[builder] -rwxr-xr-x    1 cnb      cnb            738 Apr  6 23:13 app.sh
[builder] -rw-r--r--    1 cnb      cnb            202 Apr  6 23:13 project.toml
===> EXPORTING
[exporter] no project metadata found at path './project-metadata.toml', project metadata will not be exported
[exporter] Writing tarball for layer "launcher"
[exporter] Adding layer 'launcher'
[exporter] Layer 'launcher' SHA: sha256:32608bc6d97e0193edb0360555b3e08dc6dfe1187833d8548cdd9e662213935b
[exporter] Layer 'app' SHA: sha256:4c6ff16e28deaf437a1abe5c926cdfb8afcd2c31ea544f6992ed7b665965b3f4
[exporter] Adding 1/1 app layer(s)
[exporter] Writing tarball for layer "config"
[exporter] Adding layer 'config'
[exporter] Layer 'config' SHA: sha256:9790af0612365f9299ec555ed3b78aa1d71aeca28676b9dc86ba67934f629fbc
[exporter] *** Images (sha256:1b2790f001373b10c08425ba289b0d1bb5674299dbdc27ee81562fb0121447b2):
[exporter]       index.docker.io/library/my-app-566:latest - HEAD https://index.docker.io/v2/library/my-app-566/blobs/sha256:72a4452fbca76c3f57a4894ce02659064e3aba284fa849af0ef663ed366671cf: unsupported status code 401
[exporter] 
[exporter] *** Digest: sha256:1b2790f001373b10c08425ba289b0d1bb5674299dbdc27ee81562fb0121447b2

$ ./out/pack trust-builder sample-builder
Builder sample-builder is now trusted

$ cat ~/.pack/config.toml 
[[trusted-builders]]
  name = "sample-builder"

$ ./out/pack build my-app-566 -B sample-builder -p ~/dev/buildpacks/samples/apps/bash-script/ -v --publish
Using project descriptor located at /Users/javier.romero/dev/buildpacks/samples/apps/bash-script/project.toml
Build sample-builder is trusted
Selected run image cnbs/sample-stack-run:alpine
Adding buildpack samples/bash-script version 0.0.1 to builder
Setting custom order
Saving builder with the following buildpacks:
-> samples/java-maven@0.0.1
-> samples/kotlin-gradle@0.0.1
-> samples/hello-world@0.0.1
-> samples/bash-script@0.0.1
Using build cache volume pack-cache-477bee8af0a7.build
===> CREATING
[creator] ---> DETECTING
[creator] ======== Output: samples/bash-script@0.0.1 ========
[creator] ---> Hello Bash Script buildpack
[creator] 
[creator] Here are the contents of the current working directory:
[creator] .:
[creator] total 16
[creator] drwxr-xr-x    2 cnb      cnb           4096 May 20 06:24 .
[creator] drwxr-xr-x    1 root     root          4096 May 20 06:24 ..
[creator] -rwxr-xr-x    1 cnb      cnb            738 Apr  6 23:13 app.sh
[creator] -rw-r--r--    1 cnb      cnb            202 Apr  6 23:13 project.toml
[creator] ======== Results ========
[creator] pass: samples/bash-script@0.0.1
[creator] Resolving plan... (try #1)
[creator] samples/bash-script 0.0.1
[creator] ---> ANALYZING
[creator] Previous image with name "index.docker.io/library/my-app-566:latest" not found
[creator] ---> RESTORING
[creator] ---> BUILDING
[creator] ---> Hello World buildpack
[creator] ---> Hello Bash Script buildpack
[creator] 
[creator] Here are the contents of the current working directory:
[creator] .:
[creator] total 16
[creator] drwxr-xr-x    2 cnb      cnb           4096 May 20 06:24 .
[creator] drwxr-xr-x    1 root     root          4096 May 20 06:24 ..
[creator] -rwxr-xr-x    1 cnb      cnb            738 Apr  6 23:13 app.sh
[creator] -rw-r--r--    1 cnb      cnb            202 Apr  6 23:13 project.toml
[creator] ---> EXPORTING
[creator] no project metadata found at path './project-metadata.toml', project metadata will not be exported
[creator] Writing tarball for layer "launcher"
[creator] Adding layer 'launcher'
[creator] Layer 'launcher' SHA: sha256:32608bc6d97e0193edb0360555b3e08dc6dfe1187833d8548cdd9e662213935b
[creator] Layer 'app' SHA: sha256:4c6ff16e28deaf437a1abe5c926cdfb8afcd2c31ea544f6992ed7b665965b3f4
[creator] Adding 1/1 app layer(s)
[creator] Writing tarball for layer "config"
[creator] Adding layer 'config'
[creator] Layer 'config' SHA: sha256:ed89d623eb7cf47a03403af0b6d3723a967ca4f73985b97056246193d29eb3b0
[creator] *** Images (sha256:b9279b3ca789074ae90a65492a08d60ef3300f7cb77fc7b41da03b400bcf92a3):
[creator]       index.docker.io/library/my-app-566:latest - HEAD https://index.docker.io/v2/library/my-app-566/blobs/sha256:f9f7b9d5b88c254d0988e21eec03df23a60a64d9f541b826f7d30121338240ae: unsupported status code 401
[creator] 
[creator] *** Digest: sha256:b9279b3ca789074ae90a65492a08d60ef3300f7cb77fc7b41da03b400bcf92a3
```

File: cmd/pack/main.go [hunk 1]
```diff
@@ -61,15 +61,18 @@ func main() {
...
-
-	rootCmd.AddCommand(commands.CreateBuilder(logger, cfg, &packClient))
-	rootCmd.AddCommand(commands.PackageBuildpack(logger, &packClient, buildpackage.NewConfigReader()))
 	rootCmd.AddCommand(commands.SetRunImagesMirrors(logger, cfg))
-	rootCmd.AddCommand(commands.InspectBuilder(logger, cfg, &packClient))
+
 	rootCmd.AddCommand(commands.SetDefaultBuilder(logger, cfg, &packClient))
+	rootCmd.AddCommand(commands.InspectBuilder(logger, cfg, &packClient))
 	rootCmd.AddCommand(commands.SuggestBuilders(logger, &packClient))
+	rootCmd.AddCommand(commands.TrustBuilder(logger, cfg))
```
Comment (natalieparellano) [inline cmd/pack/main.go:69]: Well that acceptance test is looking good right about now...

**Pull request merged**